contracts/test/forkMainnet/AMMWrapper/CollectFee.t.sol

```diff
@@ -1,62 +1,62 @@
-        uint256 feeFactor = 5;
+    function testCollectFeeForSwap() public {
+        uint256 feeFactor = 100;
         AMMLibEIP712.Order memory order = DEFAULT_ORDER;
         uint256 expectedOutAmount = ammQuoter.getMakerOutAmount(order.makerAddr, order.takerAssetAddr, order.makerAssetAddr, order.takerAssetAmount);
-        // Reduce `expectedOutAmount` so received amount is more than `expectedOutAmount` and
-        // the amount difference is enough for us to collect fee
-        uint256 reducedExpectedOutAmount = (expectedOutAmount * (BPS_MAX - 2 * feeFactor)) / BPS_MAX;
-        order.makerAssetAmount = reducedExpectedOutAmount;
+        // order should align with user's perspective
+        // therefore it should deduct fee from expectedOutAmount as the makerAssetAmount in order
+        uint256 fee = (expectedOutAmount * feeFactor) / BPS_MAX;
+        order.makerAssetAmount = expectedOutAmount - fee;
         bytes memory sig = _signTrade(userPrivateKey, order);
         bytes memory payload = _genTradePayload(order, feeFactor, sig);
 
         BalanceSnapshot.Snapshot memory ammWrapperMakerAsset = BalanceSnapshot.take(address(ammWrapper), order.makerAssetAddr);
 
+        vm.prank(relayer, relayer);
         userProxy.toAMM(payload);
 
-        ammWrapperMakerAsset.assertChangeGt(int256(0));
+        ammWrapperMakerAsset.assertChange(int256(fee));
     }
 
-    // Received more than expected min amount but not enough
-    function testSwapReceiveMoreThanMinOutButNotEnough() public {
-        uint256 feeFactor = 5;
+    function testFeeFactorOverwrittenWithDefault() public {
+        // set local feeFactor higher than default one to avoid insufficient output from AMM
+        uint256 feeFactor = ammWrapper.defaultFeeFactor() + 1000;
         AMMLibEIP712.Order memory order = DEFAULT_ORDER;
         uint256 expectedOutAmount = ammQuoter.getMakerOutAmount(order.makerAddr, order.takerAssetAddr, order.makerAssetAddr, order.takerAssetAmount);
-        // Reduce `expectedOutAmount` so received amount is more than `expectedOutAmount` but
-        // the amount difference is not enough for us to collect fee
-        uint256 reducedExpectedOutAmount = (expectedOutAmount * (BPS_MAX - feeFactor + 1)) / BPS_MAX;
-        order.makerAssetAmount = reducedExpectedOutAmount;
+        uint256 fee = (expectedOutAmount * feeFactor) / BPS_MAX;
+        order.makerAssetAmount = expectedOutAmount - fee;
         bytes memory sig = _signTrade(userPrivateKey, order);
         bytes memory payload = _genTradePayload(order, feeFactor, sig);
 
         BalanceSnapshot.Snapshot memory ammWrapperMakerAsset = BalanceSnapshot.take(address(ammWrapper), order.makerAssetAddr);
 
         userProxy.toAMM(payload);
 
-        ammWrapperMakerAsset.assertChangeGt(int256(0));
+        uint256 actualFee = (expectedOutAmount * ammWrapper.defaultFeeFactor()) / BPS_MAX;
+        ammWrapperMakerAsset.assertChange(int256(actualFee));
```

Comment:
I think we can replace the balance diff check with `feeFactor` param in `Swapped` event?

Reply:
Is there a way to get param in the emitted event?

Reply:
I think balance check is intuitive for me in this scenario.

Reply:
Yeah but I can add event check as well. Will do.

Reply:
Fixed in c276658.